packages/sitecore-jss-cli/src/scripts/extract-components.ts

```diff
@@ -0,0 +1,150 @@
+import chalk from 'chalk';
+import path from 'path';
+import { Argv } from 'yargs';
+import fs from 'fs';
+import * as ts from 'typescript';
+import { readConfigFile } from 'typescript';
+import { constants } from '@sitecore-jss/sitecore-jss-dev-tools';
+
+const M2M_ENDPOINT = 'https://auth-staging-1.sitecore-staging.com/oauth/token';
+
+// TODO:adjust when mesh endpoint is live
+const meshEndpoint = `${process.env.SITECORE_EDGE_URL ||
+  constants.SITECORE_EDGE_URL_DEFAULT}/api/v1/mesh`;
+
+/**
+ * @param {Argv} yargs
+ */
+export function args(yargs: Argv) {
+  return yargs.option('appFolder', {
+    requiresArg: false,
+    type: 'string',
+    describe: 'Path to app folder to get components from',
+  });
+}
+
+/**
+ * @param {Argv} yargs
+ */
+export default function builder(yargs: Argv) {
+  return yargs.command(
+    'extract-components',
+    'Reads component files and posts their code for AI learning',
+    args,
+    handler
+  );
+}
+
+/**
+ * Handler fo the extract-component API command
+ * Reads imports from the componentBuilder.ts file and posts the code to the mesh endpoint
+ * @param {object} args - The arguments passed to the command, with optional appName string
+ * @returns {Promise<void>} void
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function handler(args: any) {
+  // TODO: add alternative consent procedure when refining later
+  if (!process.env.EXTRACT_CONSENT) {
+    console.log(chalk.yellow('Skipping code extraction, EXTRACT_CONSENT is not set'));
+    return;
+  }
+  const basePath = args.appFolder ? resolveAppPath(args.appFolder) : process.cwd();
+  try {
+    const componentPaths = resolveImportFiles(basePath);
+    const bearer = await fetchBearerToken();
+    if (!bearer) {
+      console.error(chalk.red('Failed to get bearer token, aborting code extraction'));
+      return;
+    }
+
+    for (const componentPath of componentPaths!) {
+      await sendCode(componentPath, bearer);
+    }
+  } catch (error) {
+    console.error(chalk.red('Error during component extraction:', error));
+  }
+}
+
+const resolveAppPath = (appFolder: string) => {
+  if (path.isAbsolute(appFolder)) return appFolder;
+  return path.resolve(process.cwd(), appFolder);
+};
+
+export const fetchBearerToken = async () => {
+  try {
+    // TODO:adjust when M2M endpoint is live
+    const authenticateResponse = await fetch(M2M_ENDPOINT, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
content type is 'application/x-www-form-urlencoded' but body is JSON, does it work correctly?